superset/assets/src/visualizations/nvd3/NVD3Vis.js

```diff
@@ -320,25 +319,37 @@ function nvd3Vis(element, props) {
           if (tableFilter) {
             const publishedColumns = formData.publishColumns;
             let yColumn;
+            let yColumnLabel;
             let metric;
+            let key = e.series.key;
+
             if (formData.metrics.length == 1) {
               metric = formData.metrics[0];
             } else {
-              let key = e.series.key;
               metric = formData.metrics.find((metric) => {
-                return key.includes(metric.label);
+                return metric['expressionType'] ?  key.includes(metric.label) : key.includes(metric);
```

Comment:
I liked the way you have ensured to handle string metric but can we use `hasOwnProperty` or `contructor` to see metric is a string

Reply:
updated